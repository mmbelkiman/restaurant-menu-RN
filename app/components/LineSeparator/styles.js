import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  separatorWithoutMargin: {
    backgroundColor: 'gray',
    opacity: 0.2,
    height: 1,
    marginHorizontal: '0%',
  },
  separator: {
    backgroundColor: 'gray',
    opacity: 0.2,
    height: 1,
    marginHorizontal: '10%',
  },
});
