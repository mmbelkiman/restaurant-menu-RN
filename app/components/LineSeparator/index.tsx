import React from 'react';
import {View} from 'react-native';
import styles from './styles';

interface ILineSeparatorProps {
  withoutMargin?: boolean;
}

const LineSeparator: React.FC<ILineSeparatorProps> = ({withoutMargin}) => {
  return (
    <View
      style={withoutMargin ? styles.separatorWithoutMargin : styles.separator}
    />
  );
};

export default LineSeparator;
