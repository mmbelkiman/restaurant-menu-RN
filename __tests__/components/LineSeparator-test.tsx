import 'react-native';
import React from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import LineSeparator from '../../app/components/LineSeparator';

describe('LineSeparator', () => {
  const getRender = (): RenderAPI => render(<LineSeparator />);

  it('Jest snapshot', async () => {
    const json = await getRender().toJSON();
    expect(json).toMatchSnapshot();
  });
});
