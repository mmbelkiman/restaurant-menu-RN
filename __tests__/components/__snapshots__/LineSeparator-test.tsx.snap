// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineSeparator Jest snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "gray",
      "height": 1,
      "marginHorizontal": "10%",
      "opacity": 0.2,
    }
  }
/>
`;
